src/main/scala/io/ino/solrs/AsyncSolrClient.scala

```diff
@@ -609,24 +609,37 @@ class AsyncSolrClient[F[_]] protected (private[solrs] val loadBalancer: LoadBala
     val promise = futureFactory.newPromise[T]
     val startTime = System.currentTimeMillis()
 
-    val streams = requestWriter.getContentStreams(r)
-    val url = if (r.getMethod == POST && streams == null) {
-      // for POST (non-stream post), wparam should NOT be added to the url - wparam passed as form data in POST.
-      solrServer.baseUrl + getPath(r)
-    } else {
-      solrServer.baseUrl + getPath(r) + wparams.toQueryString
-    }
+    val contentWriter = requestWriter.getContentWriter(r)
+    val streams = if (contentWriter == null) requestWriter.getContentStreams(r) else null
```

Comment:
Regarding `getContentStreams`: do we need it at all? [SOLR-11657](https://issues.apache.org/jira/browse/SOLR-11657) says it should be deprecated and this is actually already done with https://github.com/apache/lucene-solr/commit/a03d6bc8c27d3d97011bc5bdc2aeb94c4820628c.

When I remove this and the related code tests are green at least...